test/monitoring/components_metrics_test.go

```diff
@@ -70,6 +70,11 @@ func testComponentsPrometheusMetrics(t *testing.T, v1api v1.API) {
 			query:         "kubelet_running_pod_count",
 			platforms:     selfHostedPlatforms,
 		},
+		{
+			componentName: "etcd",
+			query:         "etcd_server_has_leader",
+			platforms:     selfHostedPlatforms,
+		},
```

Comment:
Just realised that we can't test this on CI reliably because this metric will never be scraped by prometheus. Since it needs IP of the etcd endpoints and we only know IP after cluster is up and running.

Reply:
```
    TestPrometheus/ComponentMetrics/prometheus-etcd: components_metrics_test.go:132: no metrics found for query "etcd_server_has_leader"
    TestPrometheus/ComponentMetrics/prometheus-etcd: components_metrics_test.go:107: timed out waiting for the condition
```

Reply:
It should be the IPs of controller nodes, no?

Reply:
Yes that's correct.

Reply:
I think this test can now be re-enabled.